tests/Queue/Failed/MongoFailedJobProviderTest.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace MongoDB\Laravel\Tests\Queue\Failed;
+
+use Illuminate\Support\Facades\Date;
+use Illuminate\Support\Facades\DB;
+use MongoDB\BSON\ObjectId;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\Laravel\Queue\Failed\MongoFailedJobProvider;
+use MongoDB\Laravel\Tests\TestCase;
+use OutOfBoundsException;
+
+use function array_map;
+use function range;
+use function sprintf;
+
+class MongoFailedJobProviderTest extends TestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        DB::connection('mongodb')
+            ->collection('failed_jobs')
+            ->raw()
+            ->insertMany(array_map(static fn ($i) => [
+                '_id' => new ObjectId(sprintf('%024d', $i)),
```

Comment:
Are `_id` fields within jobs _always_ an ObjectId? I saw that MongoFailedJobProvider made some assumptions about that by using `string` for type hints and casting.

Reply:
Documents are added by the `log` method which doesn't specify `_id`. So we get an automatic ObjectId.